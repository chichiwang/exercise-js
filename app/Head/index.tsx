/* global PROJECT_URL */
import React, { useContext } from 'react';

import ManifestContext from 'app/context/ManifestContext';
import useRouteHead from 'app/hooks/useRouteHead';
import type { Manifest } from 'app/types';

function Head() {
  const manifest: Manifest = useContext(ManifestContext);
  const head = useRouteHead();

  const defaultTitle = 'Tamsui: A Universal JavaScript Application Boilerplate';
  const titleSuffix = '| Tamsui';
  const title = head?.title
    ? `${head.title} ${titleSuffix}`
    : defaultTitle;

  return (
    <head>
      <meta charSet="utf-8" />
      <title>{ title }</title>
      <meta name="author" content="Chi-chi Wang" />
      <meta
        name="keywords"
        content="Tamsui, js, JavaScript, isomorphic, Node.js, React, TypeScript, Express, React-Router, Boilerplate"
      />
      <meta name="description" content="An Express/React universal application boilerplate" />
      <meta name="robots" content="all" />
      <meta name="viewport" content="width=device-width" />
      <link rel="stylesheet" type="text/css" href={`/${manifest['app.css']}`} />
      <link rel="apple-touch-icon" sizes="180x180" href="/static/images/favicon/apple-touch-icon.png" />
      <link rel="icon" type="image/png" sizes="32x32" href="/static/images/favicon/favicon-32x32.png" />
      <link rel="icon" type="image/png" sizes="16x16" href="/static/images/favicon/favicon-16x16.png" />
      <link rel="manifest" href="/static/site.webmanifest" />
      <meta property="og:locale" content="en_US" />
      <meta property="og:image" content={`${PROJECT_URL}/static/images/site-preview.webp`} />
      <meta property="og:image:width" content="1200" />
      <meta property="og:image:height" content="630" />
      <meta property="og:image:type" content="image/webp" />
    </head>
  );
}

export default Head;
