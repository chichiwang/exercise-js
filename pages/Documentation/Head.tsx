/* global PROJECT_URL */
import React from 'react';

function Head() {
  const url = `${PROJECT_URL}/documentation`;

  return (
    <>
      <meta property="og:type" content="article" />
      <meta property="og:author" content="Chi-chi Wang" />
      <meta property="og:title" content="Developer Documentation for Tamsui: React/Express universal JavaScript boilerplate" />
      <meta property="og:description" content="Set up a universal JavaScript application on the Tamsui boilerplate." />
      <meta property="og:url" content={url} />
      <link href={url} rel="canonical" />
    </>
  );
}

export default Head;
