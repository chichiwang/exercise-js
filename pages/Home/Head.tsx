/* global PROJECT_URL */
import React from 'react';

function Head() {
  const url = `${PROJECT_URL}`;

  return (
    <>
      <meta property="og:type" content="website" />
      <meta property="og:title" content="Tamsui: A React/Express universal JavaScript boilerplate" />
      <meta property="og:description" content="Build a universal JavaScript application using React and Express." />
      <meta property="og:url" content={url} />
    </>
  );
}

export default Head;
