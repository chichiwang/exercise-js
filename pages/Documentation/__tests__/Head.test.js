import React from 'react';
import renderer from 'react-test-renderer';

import Head from '../Head';

describe('Documentation/Head component', () => {
  beforeAll(() => {
    global.PROJECT_URL = 'https://documentation.head.test';
  });

  afterAll(() => {
    delete global.PROJECT_URL;
  });

  test('matches the snapshot', () => {
    const tree = renderer.create(<Head />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
