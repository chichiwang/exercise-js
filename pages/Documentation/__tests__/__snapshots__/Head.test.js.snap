// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Documentation/Head component matches the snapshot 1`] = `
[
  <meta
    content="article"
    property="og:type"
  />,
  <meta
    content="Chi-chi Wang"
    property="og:author"
  />,
  <meta
    content="Developer Documentation for Tamsui: React/Express universal JavaScript boilerplate"
    property="og:title"
  />,
  <meta
    content="Set up a universal JavaScript application on the Tamsui boilerplate."
    property="og:description"
  />,
  <meta
    content="https://documentation.head.test/documentation"
    property="og:url"
  />,
  <link
    href="https://documentation.head.test/documentation"
    rel="canonical"
  />,
]
`;
